### ISSUE TITLE
Intermitted failure in p2p_sendtxrcncl.py

### ISSUE THREAD

**Issue by dergoegge** [bitcoin/bitcoin#26364]
I have seen `p2p_sendtxrcncl.py` fail randomly multiple times by now, e.g. : https://cirrus-ci.com/task/5513760365346816?logs=functional_tests. Also noted here: https://github.com/bitcoin/bitcoin/pull/23443#issuecomment-1281180599.

Log:

```
 node0 2022-10-21T14:09:17.893264Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\node\txreconciliation.cpp:87] [PreRegisterPeer] [txreconciliation:debug] Pre-register peer=8 
 node0 2022-10-21T14:09:17.893346Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending sendtxrcncl (14 bytes) peer=8 
 node0 2022-10-21T14:09:17.893450Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending verack (0 bytes) peer=8 
 node0 2022-10-21T14:09:17.893570Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:3363] [ProcessMessage] [net] receive version message: /python-p2p-tester:0.0.3/: version 70016, blocks=-1, us=127.0.0.1:14312, txrelay=1, peer=8 
 node0 2022-10-21T14:09:17.893649Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:3146] [ProcessMessage] [net] received: verack (0 bytes) peer=8 
 node0 2022-10-21T14:09:17.893703Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending sendcmpct (9 bytes) peer=8 
 node0 2022-10-21T14:09:17.893811Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\node\txreconciliation.cpp:145] [ForgetPeer] [txreconciliation:debug] Forget txreconciliation state of peer=8 
 node0 2022-10-21T14:09:17.893898Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending ping (8 bytes) peer=8 
 node0 2022-10-21T14:09:17.894111Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending getheaders (645 bytes) peer=8 
 node0 2022-10-21T14:09:17.894243Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:5413] [SendMessages] [net] initial getheaders (199) to peer=8 (startheight:-1) 
 node0 2022-10-21T14:09:17.894355Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending feefilter (8 bytes) peer=8 
 node0 2022-10-21T14:09:17.894489Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:3146] [ProcessMessage] [net] received: ping (8 bytes) peer=8 
 node0 2022-10-21T14:09:17.894554Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:2775] [PushMessage] [net] sending pong (8 bytes) peer=8 
 node0 2022-10-21T14:09:17.894682Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:3146] [ProcessMessage] [net] received: wtxidrelay (0 bytes) peer=8 
 node0 2022-10-21T14:09:17.894740Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:3460] [ProcessMessage] [net] wtxidrelay received after verack from peer=8; disconnecting 
 test  2022-10-21T14:09:17.954000Z TestFramework.p2p (DEBUG): Closed connection to: 127.0.0.1:14312 
 node0 2022-10-21T14:09:17.954581Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net.cpp:572] [CloseSocketDisconnect] [net] disconnecting peer=8 
 node0 2022-10-21T14:09:17.954849Z [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\src\net_processing.cpp:1528] [FinalizeNode] [net] Cleared nodestate for peer=8 
 test  2022-10-21T14:09:17.970000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\test_framework.py", line 133, in main
                                       self.run_test()
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\p2p_sendtxrcncl.py", line 138, in run_test
                                       peer.send_and_ping(msg_verack())
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\p2p.py", line 553, in send_and_ping
                                       self.sync_with_ping(timeout=timeout)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\p2p.py", line 570, in sync_with_ping
                                       self.wait_until(test_function, timeout=timeout)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\p2p.py", line 463, in wait_until
                                       wait_until_helper(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\util.py", line 267, in wait_until_helper
                                       if predicate():
                                     File "C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\test\functional\test_framework\p2p.py", line 460, in test_function
                                       assert self.is_connected
                                   AssertionError
```

**Comment by MarcoFalke** [comment]
Fixed in https://github.com/bitcoin/bitcoin/pull/26381, maybe

**Comment by brunoerg** [comment]
https://cirrus-ci.com/task/5669452258738176?logs=ci#L3971
https://cirrus-ci.com/task/6689799049314304?logs=ci#L4248

**Comment by mzumsande** [comment]
Possible explanation: https://github.com/bitcoin/bitcoin/pull/26396#discussion_r1012501223 - I think using a separate `p2p_idx` for each `add_outbound_p2p_connection` would fix this. It is late here, will open a PR tomorrow if no one else wants to do it before.

**Comment by mzumsande** [comment]
Fixed in #26448
